phylogeny/prototypeSelection.py

```diff
@@ -511,7 +519,9 @@ def prototype_selection_constructive_maxdist(dm: DistanceMatrix,
     # the element to remove first is the one that has smallest distance to all
     # other. "Removing" works by tagging its distance-sum as infinity. Plus, we
     # decrease the number of available elements by one.
-    minElmIdx = currDists.argmin()
+    minElmIdx = next(x for x in currDists.argsort()
```

Comment:
@sjanssen2 here are the benchmark result: with the test dataset you already implemented in the unit test:

2 cores, 5 prototypes from 20 elememts:
no-core: 5.4494, filter-while-loop: 5.3082, filter-then-loop: 5.2166

3 cores, 10 prototypes from 100 elements:
no-core: 26.8818, filter-while-loop: 26.7458, filter-then-loop: 26.5841

The result suggests that filter-then-loop is a fast heuristic solution with the cost of reduced sum of distances.